src/abstractdataframe/join.jl

```diff
@@ -76,21 +89,127 @@ function join_idx(left, right, max_groups)
      right_sorter[right_indexer], right_sorter[rightonly_indexer])
 end
 
-function DataArrays.PooledDataVecs(df1::AbstractDataFrame,
-                                   df2::AbstractDataFrame)
+# FIXME: rename this, and possibly move to CategoricalArrays.jl
+function PooledDataVecs{S,N}(v1::NullableCategoricalArray{S,N},
```

Comment:
Yeah, I'm not sure about the best name. Maybe `join!`. Ideally, we would also avoid accessing the `refs` field from the `PooledDataVecs(::AbstractDataFrame, ::AbstractDataFrame)` method below. But I'm even less sure what the API should look like in CategoricalArrays.jl. Could still improve this in steps.